x/nft/client/rest/query.go

```diff
@@ -57,6 +58,57 @@ func registerQueryRoutes(cliCtx client.Context, r *mux.Router) {
 
 	// Query if an address is an authorized operator for another address
 	r.HandleFunc(fmt.Sprintf("/%s/isApprovedForAll", types.ModuleName), queryIsApprovedForAll(cliCtx)).Methods("POST")
+
+	// Query for collections by denomIds
+	r.HandleFunc(fmt.Sprintf("/%s/collectionsByDenomIds", types.ModuleName), queryCollectionsByDenomIds(cliCtx)).Methods("POST")
+}
+
+// Get the collections by denom ids with their nfts
+func queryCollectionsByDenomIds(cliCtx client.Context) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		var req queryCollectionsByDenomIdsRequest
+		if !rest.ReadRESTReq(w, r, cliCtx.LegacyAmino, &req) {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, "failed to parse request")
+			return
+		}
+
+		for _, id := range req.DenomIds {
+			if err := types.ValidateDenomID(id); err != nil {
+				rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
```

Comment:
Missing return, I see a few other similar cases in the already committed REST methods.

Reply:
fixed

Reply:
Also, please if you can pinpoint exactly where the returns are missing